ddclient

```diff
@@ -2306,6 +2305,16 @@ sub get_ip {
     return $ip;
 }
 
+######################################################################
+## is_ipv6() validates if string is valid IPv6 address
+######################################################################
+sub is_ipv6 {
+    my $value = shift;
+    # This little gem from http://home.deds.nl/~aeron/regex/
+    return undef if $value !~ /^(((?=.*(::))(?!.*\3.+\3))\3?|([\dA-F]{1,4}(\3|:\b|$)|\2))(?4){5}((?4){2}|(((2[0-4]|1\d|[1-9])?\d|25[0-5])\.?\b){4})\z/ai;
+    return $value
```

Comment:
Why not just return the result of the bind? Like this:

```suggestion
    return $value !~ /^(((?=.*(::))(?!.*\3.+\3))\3?|([\dA-F]{1,4}(\3|:\b|$)|\2))(?4){5}((?4){2}|(((2[0-4]|1\d|[1-9])?\d|25[0-5])\.?\b){4})\z/ai;
```

Reply:
It depends if we want to replicate the behavior of `is_ipv6` in Data::Validate:IP or do our own thing.  If replicating then the function should return `undef` or the valid IPv6 as a string, which is what I implemented.  The actual code in `is_ipv6` does this...
```
## no critic (RegularExpressions::ProhibitCaptureWithoutTest)
$value =~ /(.+)/;
return $1;
```
Which I don't really understand why that is necessary (I am not a Perl expert).  Do you know?

However if we want to return a true/false (which is perfectly fine with me, and I think more correct) then we change to your suggestion... though `!~` needs to change to` =~` 

Let me know what you prefer (I vote to change)

David

Reply:
Let's change it.

> though !~ needs to change to =~

Oops, yup.

Reply:
Actually, I noticed that there's a capture group around the whole IPv6 address, so how about this:
```suggestion
    $value =~ /^(((?=.*(::))(?!.*\3.+\3))\3?|([\dA-F]{1,4}(\3|:\b|$)|\2))(?4){5}((?4){2}|(((2[0-4]|1\d|[1-9])?\d|25[0-5])\.?\b){4})\z/ai;
    return $1;
```
It's one line longer, but it has the same behavior as `is_ipv6`.

Reply:
Ugh, nevermind. The RE is awfully complicated and I don't think I adjusted it properly (that `(?4)` might need to be `(?5)`, not sure). Probably best to just return bool.

Reply:
I went ahead and made the change.